pkg/cmd/repo/default/default.go

```diff
@@ -0,0 +1,250 @@
+package base
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"sort"
+	"strings"
+
+	ctx "context"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/context"
+	"github.com/cli/cli/v2/git"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type iprompter interface {
+	Select(string, string, []string) (int, error)
+}
+
+type DefaultOptions struct {
+	IO         *iostreams.IOStreams
+	Remotes    func() (context.Remotes, error)
+	HttpClient func() (*http.Client, error)
+	Prompter   iprompter
+
+	Repo     ghrepo.Interface
+	ViewMode bool
+}
+
+func NewCmdDefault(f *cmdutil.Factory, runF func(*DefaultOptions) error) *cobra.Command {
+	opts := &DefaultOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Remotes:    f.Remotes,
+		Prompter:   f.Prompter,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "default [<repository>]",
+		Short: "Configure default repository",
+		Long: heredoc.Doc(`
+			Set default repository for current directory.
```

Comment:
Nit: adding definite articles could help readability. Also, when saying "repository" in a context where it's ambiguous whether it is a remote or a local repository, it might be a good idea to provide clarifications.

```suggestion
			Set the default remote repository for the current directory.
```